goaway.go

```diff
@@ -121,17 +121,17 @@ func (g *ProfanityDetector) ExtractProfanity(s string) string {
 	s, _ = g.sanitize(s, false)
 	// Check for false negatives
 	for _, word := range g.falseNegatives {
-		if match := strings.Contains(s, word); match {
+		if match := strings.Contains(s, strings.ToLower(word)); match {
 			return word
 		}
 	}
 	// Remove false positives
 	for _, word := range g.falsePositives {
-		s = strings.Replace(s, word, "", -1)
+		s = strings.Replace(s, strings.ToLower(word), "", -1)
 	}
 	// Check for profanities
 	for _, word := range g.profanities {
-		if match := strings.Contains(s, word); match {
+		if match := strings.Contains(s, strings.ToLower(word)); match {
 			return word
 		}
 	}
```

Comment:
This is very inefficient; it's not something I can really accept.

Let me offer you an alternative solution: instead of applying `strings.ToLower` in `ExtractProfanity`, we should call it when `WithCustomDictionary` is called.

Ideally, the slices passed to `WithCustomDictionary` should all be lowercased, but it doesn't hurt to lowercase them on initialization, especially because go-away itself expects the dictionaries to be lowercased.

Reply:
This is a good point and an acceptable request I have committed a function that is now called within WithCustomDictionary that takes the submitted slice of profanities and returns the list run through the ToLower function.